src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/FixAllContextHelper.cs

```diff
@@ -50,6 +54,49 @@ internal static class FixAllContextHelper
 
                     break;
 
+                case FixAllScope.ContainingMember or FixAllScope.ContainingType:
+                    if (document != null && !await document.IsGeneratedCodeAsync(cancellationToken).ConfigureAwait(false))
+                    {
+                        if (!fixAllSpans.IsEmpty)
+                        {
+                            // We are processing a span-based FixAllContext created for a partial declaration.
+                            var documentDiagnostics = ImmutableArray<Diagnostic>.Empty;
+                            foreach (var fixAllSpan in fixAllSpans)
+                            {
+                                documentDiagnostics = documentDiagnostics.AddRange(await fixAllContext.GetDocumentSpanDiagnosticsAsync(
+                                    document, fixAllSpan).ConfigureAwait(false));
+                            }
+
+                            return ImmutableDictionary<Document, ImmutableArray<Diagnostic>>.Empty.SetItem(document, documentDiagnostics);
+                        }
+                        else if (triggerSpan.HasValue &&
+                            document.GetLanguageService<IFixAllSpanMappingService>() is IFixAllSpanMappingService spanMappingService)
+                        {
+                            // We need to compute diagnostics for each of the containing member/type and its partial declarations
+                            // using the trigger span.
+                            var documentsAndSpans = await spanMappingService.GetFixAllSpansAsync(document,
+                                triggerSpan.Value, fixAllContext.Scope, fixAllContext.CancellationToken).ConfigureAwait(false);
+                            var unused = PooledDictionary<Document, ImmutableArray<Diagnostic>>.GetInstance(out var builder);
+                            foreach (var (documentToFix, spans) in documentsAndSpans)
+                            {
+                                foreach (var span in spans)
+                                {
+                                    var documentDiagnostics = await fixAllContext.GetDocumentSpanDiagnosticsAsync(documentToFix, span).ConfigureAwait(false);
+                                    if (builder.TryGetValue(documentToFix, out var existingDiagnostics))
+                                    {
+                                        documentDiagnostics = existingDiagnostics.AddRange(documentDiagnostics);
```

Comment:
i don't get what's happening here.  if we just computed the doc span diagnostics 3 lines above... why are we adding other diagnostics to it?

Reply:
Should hopefully be much easier to follow after the refactoring.